source/common/config/utility.h

```diff
@@ -269,6 +268,30 @@ class Utility {
     return config;
   }
 
+  /**
+   * Translate the typed any field into a proto message provided by the implementation factory.
+   * @param typed_config typed configuration.
+   * @param validation_visitor message validation visitor instance.
+   * @param factory implementation factory with the method 'createEmptyConfigProto' to produce a
+   * proto to be filled with the translated configuration.
+   */
+  template <class Factory>
+  static ProtobufTypes::MessagePtr
+  translateAnyToFactoryConfig(const ProtobufWkt::Any& typed_config,
```

Comment:
Thanks, this looks clean. Are there some existing sites that we should refactor to use this?

Reply:
I didn't do a thorough search but I guess existing proto should include the deprecated config field. Otherwise, this function should exist for converting the typed config to factory configuration. 

We could replace all the `translateToFactoryConfig` with `translateAnyToFactoryConfig` because the config field is deprecated. It will break the compatibility.